daft/io/_csv.py

```diff
@@ -21,13 +21,14 @@
 def read_csv(
     path: Union[str, List[str]],
     schema_hints: Optional[Dict[str, DataType]] = None,
+    schema_override: Optional[Dict[str, DataType]] = None,
```

Comment:
I'm wondering if an `infer_schema` boolean flag dictating the semantics for the `schema_hints` arg (renaming it to `schema`) makes more sense than having an additional `schema_override` arg, since `schema_hints` and `schema_override` can't both be provided so having both is a bit redundant. Also, our semantics for `schema_override` is flipped compared to polars (used as a partially-specified column dtype override map during their schema inference), which might be confusing for users.

Survey of the APIs for other libraries:
- pyarrow has a `column_types: Dict[str, DataType]` arg as a fully-specified schema that disables schema inference, with no hint mechanism: https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html#pyarrow.csv.ConvertOptions
- polars has both `schema` as a fully-specified schema that disables schema inference, and `schema_override` as a partially-specified column dtype override map during their schema inference: https://docs.pola.rs/py-polars/html/reference/api/polars.read_csv.html
- pandas has a `dtype` arg that is a partially-specified schema with no effect on schema inference: https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html
- pyspark has a `schema` arg and `inferSchema` and `enforceSchema` args: https://spark.apache.org/docs/3.1.3/api/python/reference/api/pyspark.sql.DataFrameReader.csv.html

My vote would be:
```python
schema: Optional[Dict[str, DataType]] = None,
infer_schema: bool = True,
```
similar to the pyspark semantics, where `schema` is a fully-specified schema that disables schema inference if `infer_schema == False`, otherwise it's a partially-specified schema that is applied after schema inference. We could keep `schema_hints` for backwards compatibility (with a deprecation warning) and internally map it to `schema=schema_hints` and `infer_schema=True`.

@samster25 any thoughts here?

Reply:
I agree having the two different schema args can be confusing, and I like your suggestion! I guess the only edge case here would be if `schema = None` and `infer_schema = False` then we'd throw an error?